contracts/index_executor/src/lib.rs

```diff
@@ -159,15 +150,20 @@ mod index_executor {
             Ok(())
         }
 
-        /// Sets the graph, callable only to a specifically crafted management tool,
-        /// should not be called by anyone else
         #[ink(message)]
-        pub fn set_graph(&mut self, graph: RegistryGraph) -> Result<()> {
-            // self.ensure_owner()?;
-            self.graph = TryInto::<Graph>::try_into(graph)
-                .or(Err(Error::SetGraphFailed))?
-                .encode();
-            Self::env().emit_event(GraphSet {});
+        pub fn update_registry(
+            &mut self,
+            chain: String,
+            endpoint: String,
+            indexer: String,
+        ) -> Result<()> {
+            self.ensure_owner()?;
+
+            if let Some(index) = self.registry.chains.iter().position(|x| x.name == chain) {
```

Comment:
Return an error if not found? Otherwise you cannot know if the change is applied or not.

Reply:
Done